trpcultivate_phenotypes/src/Plugin/Validators/ProjectGenusMatch.php

```diff
@@ -154,4 +177,107 @@ public function validateMetadata(array $form_values) {
     ];
   }
 
+  /**
+   * Process failed validation from ProjectGenusMatch into a render array.
+   *
+   * @param array $failure
+   *   An associative array that was returned by the ProjectGenusMatch validator
+   *   in the event of failed validation. It contains the following keys:
+   *   - 'case': a developer-focused string describing the case checked.
+   *   - 'valid': FALSE to indicate that validation failed.
+   *   - 'failedItems': an array of items that failed with the following keys.
+   *     - 'project_provided': The name of the project provided.
+   *     - 'genus_provided': The name of the genus provided.
+   * @param array $tokens
+   *   [OPTIONAL] An array of values to use for token replacement.
+   *   The following tokens can be specfied as keys, with value as the
+   *   replacement value for the token. These apply to all failure cases.
+   *   - 'project': replaces the word "project".
+   *   - 'contact-admin': replaces the phrase "contact your administrator".
+   *   The following token keys will substitute the entire existing case message
+   *   with the value of that token.
+   *   - 'case-message-1': "Project does not exist"
+   *   - 'case-message-2': "Project has no genus set and could not compare with
+   *     the genus provided"
+   *   - 'case-message-3': "Genus does not match a genus set to the project".
+   *
+   * @return array
+   *   A render array of type unordered list which is used to display feedback
+   *   to the user about the case that failed and the failed items from the
+   *   input file. Each item in the list contains either the project name or the
+   *   genus, whichever one caused the failure.
```

Comment:
Technically both project/genus are included for some cases :thinking: plus it's actually and item containing an unordered list. What about something like:

> A render array used to display feedback to the user about the failure. Specifically, we return an item where the title describes the case triggered and the value is an unordered list providing the genus and/or project name (each as their own list item). The items in the list are chosen based on the case in order to provide applicable context to the failure.

Reply:
I broke down your suggested text into bullet points in commit https://github.com/TripalCultivate/TripalCultivate-Phenotypes/pull/159/commits/c4583ed2de1ff495adc9723f431e59f7c87a3cb9. Please let me know if you like it!